### PR TITLE
Fix for Issue - 146

### LINKED ISSUE #146: "How do I know if I need to update" section should mention auto-notifications
https://tessel.gitbooks.io/t2-docs/content/API/CLI.html#how-do-i-know-if-i-need-to-update-my-t2

The "How do I know..." section linked above should mention what you will get automatic updates about. Specifically, it should mention that:

* The CLI will tell you if you are running an old version of CLI if you are connected to the internet and have a version newer than (when did this get introduced?) when you run any t2 command
* The CLI will not tell you if you are running an old version of firmware

### PR DESCRIPTION
**Pull request by mittalshravika**
Fixes #146. Updated the "How do I know if I need to update" section in t2-docs with update instructions for T2-CLI and T2-Firmware.

### CODE REVIEW

File: API/CLI.md [hunk 1]
```diff
@@ -15,6 +15,20 @@ The best place to go next is [the Tessel 2 start experience](http://tessel.io/t2
 ## Updating Tessel 2 On-board OS/Firmware
 
 ### How do I know if I need to update my T2?
+T2 - CLI and T2 - Firmware are separately versioned.
```
Comment (rwaldron) [inline API/CLI.md]: This spacing makes it hard to read, and it won't be control-f search friendly for users that would spell it out as it appears in the respective projects, ie: "t2-cli" and "t2-firmware"

Comment (mittalshravika) [inline API/CLI.md]: Sure. Will make the changes.

File: API/CLI.md [hunk 2]
```diff
@@ -15,6 +15,20 @@ The best place to go next is [the Tessel 2 start experience](http://tessel.io/t2
 ## Updating Tessel 2 On-board OS/Firmware
 
 ### How do I know if I need to update my T2?
+t2-cli and t2-firmware are separately versioned.
+
+* For updating t2-cli:
+
+  If the computer running the t2-cli is connected to the internet, the CLI checks whether it is up to date or not. If it is not, the CLI will provide the update instructions using a notifier to the user (as shown below). It is to be updated with a global npm re-install of the CLI.
```
Comment (Frijol) [inline API/CLI.md]: At the end, please specify the specific command: `npm install -g t2-cli`

Comment (Frijol) [inline API/CLI.md]: "If it is not" leaves "it" ambiguous– the computer? The CLI? Suggested rephrase: "When a CLI command is run on an internet-connected computer, the CLI automatically checks for updates to the `t2-cli` module. If an update is available..."

Comment (Frijol) [inline API/CLI.md]: "It is to be updated" also leaves the "it" a bit ambiguous. Suggestion: "The CLI can be updated with.."

File: API/CLI.md [hunk 3]
```diff
@@ -15,6 +15,20 @@ The best place to go next is [the Tessel 2 start experience](http://tessel.io/t2
...
+
+  _The CLI only checks once in a day ensuring that the CLI functioning does not slow down._
```
Comment (Frijol) [inline API/CLI.md]: substitute: "once in a day ensuring" with "once per day to ensure"

File: API/CLI.md [hunk 4]
```diff
@@ -15,6 +15,20 @@ The best place to go next is [the Tessel 2 start experience](http://tessel.io/t2
...
+
+  <img src="/images/CLI_Update.png" width="400">
+
+* For updating t2-firmware:
+
+  The user does not get update information for t2-firmware from the CLI. The most convenient and fastest way to find is by running the `t2 update` command.
```
Comment (Frijol) [inline API/CLI.md]: Missing a phrase, "way to find is" should be "way to find out if a firmware update is needed is"

File: API/CLI.md [hunk 5]
```diff
@@ -15,6 +15,20 @@ The best place to go next is [the Tessel 2 start experience](http://tessel.io/t2
...
+
 The easiest way is to run `t2 update` and it will update automatically if there is a newer firmware version available. Otherwise, you can run `t2 version` to get the version running on your Tessel, and then `t2 update -l` to see the 10 newest versions available.
```
Comment (Frijol) [inline API/CLI.md]: This one needs tabbing in or it won't render as part of the bullet point

Comment (Frijol) [inline API/CLI.md]: This bit seems to be partially redundant with the paragraph above it

Comment (mittalshravika) [inline API/CLI.md]: Sure. Will make all the changes suggested

**Comment by mittalshravika** [comment]
I have made the changes suggested.

**Comment by Frijol** [comment]
Looks good!

**Pull request merged**